### ISSUE TITLE
Unnecessary warnings or bug?

### ISSUE THREAD

**Issue by 4ntoine** [avr-llvm/lld#3]
Should it be `&` or `&&`?

Warnings during build:

```
/Users/asmirnov/Documents/dev/src/avr-llvm/llvm/tools/lld/lib/ReaderWriter/ELF/AVR/AVRRelocationHandler.cpp:160:31: warning: 
      use of logical '&&' with constant operand [-Wconstant-logical-operand]
  uint64_t result = (((target && (3<<4)) << 2) | // MSB
                              ^  ~~~~~~
/Users/asmirnov/Documents/dev/src/avr-llvm/llvm/tools/lld/lib/ReaderWriter/ELF/AVR/AVRRelocationHandler.cpp:160:31: note: 
      use '&' for a bitwise operation
  uint64_t result = (((target && (3<<4)) << 2) | // MSB
                              ^~
                              &
/Users/asmirnov/Documents/dev/src/avr-llvm/llvm/tools/lld/lib/ReaderWriter/ELF/AVR/AVRRelocationHandler.cpp:160:31: note: 
      remove constant to silence this warning
  uint64_t result = (((target && (3<<4)) << 2) | // MSB
                             ~^~~~~~~~~
/Users/asmirnov/Documents/dev/src/avr-llvm/llvm/tools/lld/lib/ReaderWriter/ELF/AVR/AVRRelocationHandler.cpp:161:31: warning: 
      use of logical '&&' with constant operand [-Wconstant-logical-operand]
                     ((target && 0xf)));         // LSB
                              ^  ~~~
/Users/asmirnov/Documents/dev/src/avr-llvm/llvm/tools/lld/lib/ReaderWriter/ELF/AVR/AVRRelocationHandler.cpp:161:31: note: 
      use '&' for a bitwise operation
                     ((target && 0xf)));         // LSB
                              ^~
                              &
/Users/asmirnov/Documents/dev/src/avr-llvm/llvm/tools/lld/lib/ReaderWriter/ELF/AVR/AVRRelocationHandler.cpp:161:31: note: 
      remove constant to silence this warning
                     ((target && 0xf)));         // LSB
```

**Comment by dylanmckay** [comment]
You're right. I have fixed this and will push it tomorrow.

**Comment by 4ntoine** [comment]
Good

**Comment by dylanmckay** [comment]
Closed in 8f5afe1d90ac97a835ef569679a3e7ffce30cede
